Kernel/semaphores.c

```diff
@@ -113,7 +112,7 @@ TSem sem_open(const char * name, uint8_t initialValue) {
         return SEM_FAILED;
     }
 
-    if(rnm_nameResource(namer, semaphores[i], name, &(semaphores[i]->name))!=0){
+    if(rnm_nameResource(namer,(void *)(int64_t)i, name, &(semaphores[i]->name))!=0){
         //check what should be freed
```

Comment:
```suggestion 
wq_free(semaphores[i]->waitingProcesses);
mm_free(semaphores[i]);
semaphores[i] = NULL;
```

I think that this will be enough